oaknational/Oak-Web-Application
# Fix/hubspot

## Description

- adds `getAnonymousId` function which looks for the old anonymous-id or generates a new one
- updates consent-manager to have `consentState = "pending" | "enabled" | "disabled"`
  - this allows us to queue events in the case of "pending"
- adds `withQueue` function which wraps analytics services in a queue
- call `identify` function on form submission

## Issue(s)

Fixes #376 

## How to test

1. Go to {cloud link}
2. Click about
3. Complete sign up form
4. Go to https://app-eu1.hubspot.com/contacts/24947711/objects/0-1/views/all/list
5. You should see your contact created in hubspot

## Checklist
